D1 REST API latency changelog (#22751)
cloudflare/cloudflare-docs: src/content/changelog/d1/2025-05-30-d1-rest-api-latency.mdx

---
title: Faster D1 REST API
description: D1 REST API request latency improvements.
products:
  - d1
  - workers
date: 2025-05-29T00:00:00Z
---

> Users using Cloudflare's [REST API](/api/resources/d1/) to query their D1 database can see lower end-to-end request latency now that D1 authentication is performed at the closest Cloudflare network data center that recieved the request. Previously, authentication required D1 REST API requests to proxy to Cloudflare's core centralized centers, which added network round trips. 

Reviewer: ```suggestion
Users using Cloudflare's [REST API](/api/resources/d1/) to query their D1 database can see lower end-to-end request latency now that D1 authentication is performed at the closest Cloudflare network datacenter that received the request. Previously, authentication required D1 REST API requests to proxy to Cloudflare's core centralized centers, which added network round trips. 
```

> Users using Cloudflare's [REST API](/api/resources/d1/subresources/database/methods/query/) to query their D1 database can see lower end-to-end request latency now that D1 authentication is performed at the Cloudflare network data center hosting your database. Previously, authentication required D1 REST API requests to proxy to Cloudflare's core data centers, which added network round trips to select, central locations. 

Author: accepted

> D1 query endpoints like `/query` and `/raw` have the most noticeable improvements since they no longer access Cloudflare's core data centers. D1 control plane endpoints such as those to create and delete databases see smaller improvements, since they still require access to Cloudflare's core data centers for other control plane metadata.

Author: CF website uses 2 words so I'm going to keep as is

> D1 query endpoints like `/query` and `/raw` have the most noticeable improvements since they no longer access Cloudflare's core data centers. D1 control plane endpoints such as those to create and delete databases see smaller improvements, due to authentication at the network data center hosting a database, but still require access to Cloduflare's core data centers.

Reviewer: ```suggestion
D1 query endpoints like `/query` and `/raw` have the most noticeable improvements since they no longer access Cloudflare's core data centers. D1 control plane endpoints such as those to create and delete databases see smaller improvements, since they still require access to Cloudflare's core data centers for other control plane metadata.
```

Author: accepted

> Performance improvements apply to only the REST API. Users can now expect similar performance querying their database via the REST API and via a [Worker](/d1/best-practices/query-d1/#query-d1-with-workers-binding-api).

Author: @lambrospetrou Are there any plans to address the REST overhead? Is the overhead 10s of ms (if so I could still frame as comparable performance which is the goal)

> Performance improvements apply to only the REST API. Users can now expect similar performance querying their database via the REST API and via a [Worker](/d1/best-practices/query-d1/#query-d1-with-workers-binding-api).

Reviewer: It's 100-300ms more than the Worker Binding depending on caching of downstream services, so not negligible in some cases.

> Performance improvements apply to only the REST API. Users can now expect similar performance querying their database via the REST API and via a [Worker](/d1/best-practices/query-d1/#query-d1-with-workers-binding-api).

Author: > It's 100-300ms more than the Worker Binding depending on caching of downstream services

none of this is specific to just D1? applies to any CF API usage?
